--- /tmp/nxt_python.c	Thu Sep  8 11:31:17 2022
+++ nxt_python.c	Mon May 30 18:34:34 2022
@@ -118,6 +118,11 @@
             size = (len + 1) * sizeof(wchar_t);
         }
 
+	 PyConfig config;
+	 PyConfig_InitIsolatedConfig(&config);
+	 Py_InitializeFromConfig(&config);
+	 PyConfig_Clear(&config);
+
         nxt_py_home = nxt_malloc(size);
         if (nxt_slow_path(nxt_py_home == NULL)) {
             nxt_alert(task, "Failed to allocate memory");
@@ -125,13 +130,13 @@
         }
 
         if (pep405) {
-            mbstowcs(nxt_py_home, c->home, len);
-            mbstowcs(nxt_py_home + len, bin_python, sizeof(bin_python));
-            Py_SetProgramName(nxt_py_home);
+            //mbstowcs(nxt_py_home, c->home, len);
+            //mbstowcs(nxt_py_home + len, bin_python, sizeof(bin_python));
+            //Py_SetProgramName(nxt_py_home);
 
         } else {
-            mbstowcs(nxt_py_home, c->home, len + 1);
-            Py_SetPythonHome(nxt_py_home);
+            //mbstowcs(nxt_py_home, c->home, len + 1);
+            //Py_SetPythonHome(nxt_py_home);
         }
 
 #else
